sed/loader/flash/loader.py

```diff
@@ -0,0 +1,589 @@
+
+    def fill_na(
+        self, dataframes: List,
+    ) -> dd.DataFrame:  # pylint: disable=no-self-use
+        """Routine to fill the NaN values with intrafile forward filling."""
+        channels: List[str] = self.channels_per_pulse + self.channels_per_train
+        for i, _ in enumerate(dataframes):
+            dataframes[i][channels] = dataframes[i][channels].fillna(
+                method="ffill",
+            )
+
+        # This loop forward fills between the consective files.
+        # The first run file will have NaNs, unless another run
+        # before it has been defined.
+        for i in range(1, len(dataframes)):
+            # Take only pulse channels
+            subset = dataframes[i][channels]
+            # Find which column(s) contain NaNs.
+            is_null = subset.loc[0].isnull().values.compute()
+            # Statement executed if there is more than one NaN value in the
+            # first row from all columns
+            if is_null.sum() > 0:
+                # Select channel names with only NaNs
+                channels_to_overwrite = list(compress(channels, is_null[0]))
+                # Get the values for those channels from previous file
+                values = dataframes[i - 1][channels].tail(1).values[0]
+                # Extract channels_to_overwrite from values and create a dictionary
+                fill_dict = dict(zip(channels, values))
+                fill_dict = {
+                    k: v
+                    for k, v in fill_dict.items()
+                    if k in channels_to_overwrite
+                }
+                # Fill all NaNs by those values
+                dataframes[i][channels_to_overwrite] = subset[
+                    channels_to_overwrite
+                ].fillna(fill_dict)
+
+        return dd.concat(dataframes)
+
+    # pylint: disable=R0201
+    def parse_metadata(
+        self,
+        files: Sequence[str],  # pylint: disable=unused-argument
+    ) -> dict:
+        """Dummy
+
+        Args:
+            files (Sequence[str]): _description_
+
+        Returns:
+            dict: _description_
+        """
+        return {}
+
+    def get_count_rate(
+        self,
+        fids: Sequence[int] = None,
+        **kwds,
+    ):
+        return None, None
+
+    def get_elapsed_time(self, fids=None, **kwds):
+        return None
+
+    def read_dataframe(
+        self,
+        files: Sequence[str] = "",
+        folder: str = "",
+        ftype: str = "h5",
+        **kwds,
+    ) -> Tuple[dd.DataFrame, dict]:
+        """Read express data from DAQ, generating a parquet in between."""
+        # create a per_file directory
+        temp_parquet_dir = self.data_parquet_dir.joinpath("per_file")
+        os.makedirs(temp_parquet_dir, exist_ok=True)
+
+        folder = str(self.data_raw_dir)
+        # Prepare a list of names for the files to read and parquets to write
+        runs = files if isinstance(files, list) else [files]
```

Comment:
You "misuse" the "files" parameter, and interprete it as "runs". I would discourage this, because this is not the expected behaviour compared to the other readers. There, given a list of files, it loads these, given a folder, it loads all compatible files in this folder. Now, you want to load all files from a specific run. So I would suggest to add another parameter "runs: Sequence(int)" that you use for this. then you can still implement the other two functionalities, in addition to this one. You can just pass it along to the reader through the kwds arguments (maybe we need to extend the clause in the init function that calls the load function)